### ISSUE TITLE
NOTICE: recent breaking changes with timing for some Custom Inject Functions

### ISSUE THREAD

**Issue by nartc** [angular-threejs/angular-three#45]
Hi folks, I'm posting this as an issue to get folks' attention. 

Technically, since we're still in beta, there's really no **breaking change** but I feel responsible for letting people know this change.

### Context

There are various Custom Inject Functions (functions that start with `inject***` and make use of APIs that rely on Injection Context) in Angular Three (i.e: `injectLoader`, `injectAnimations`, `injectFBO` etc...). 

These functions use `afterNextRender` **INTERNALLY** to assert certain things from a timing perspective (eg: after first render has finished) to proceed with their own logic.

### Change

Latest release (`2.0.0-beta.314`) removes ALL internal `afterNextRender`. The timing control should be the consumers' responsibility rather than the library. Please check the following commits and their messages for more details

https://github.com/angular-threejs/angular-three/commit/d57d1cc9503abc61ae953a841991130953ac412a
https://github.com/angular-threejs/angular-three/commit/537be01011c162c36c6b22fbad1fe59abd4ea9c9
https://github.com/angular-threejs/angular-three/commit/0599c1a9329dfd3c0e6653952e1670b3e6cedbb0
https://github.com/angular-threejs/angular-three/commit/47dba5712f8f6e5c978dab356daa4f9bca073e6f
https://github.com/angular-threejs/angular-three/commit/c325c9296650a812939e72d3bd98bcb143b72345

TLDR; control the timing by using `Signal<FnParameterType | SomeFalsyType>` (`SomeFalsyType` can be `undefined`, `null`, or `undefined | null`, or literally anything that would pass the JavaScript falsy check) or use other timing fn like `afterNextRender` (and its phases).

### Exceptions

`angular-three-cannon` is the current exception to this change. `injectBody` and `injectConstraint` still use `afterNextRender` internally. This is probably a soft blocker for v2 unfortunately.

Please let me know if you have any question at all
